### DISCUSSION TITLE
download project to plc

### DISCUSSION THREAD

**Discussion by Amy15Fee** [ottowayi/pycomm3#258] [Q&A]
Hello,

i just wanted to ask if it is possible to download a project to the plc with this python wrapper?
I'm trying to write some tests and this feature would help a lot to automate the tests later on.

**Comment by ottowayi** [comment]
no, that is proprietary Rockwell functionality that is impossible outside of their software.  Although, I do have a copy of a utility that is a CLI for Logix.  I haven't used it for downloading, but I did use it to script automated uploads and exports to L5X files.  It does require you to have the correct version of Logix for the project you want to download installed, but as far as I know it still works with the latest versions.  If you want to send me an email (found on my profile), I can send you a copy of it.
